src/index.js

```diff
@@ -16,6 +16,22 @@ const pluginName = 'mini-css-extract-plugin';
 const REGEXP_CHUNKHASH = /\[chunkhash(?::(\d+))?\]/i;
 const REGEXP_CONTENTHASH = /\[contenthash(?::(\d+))?\]/i;
 const REGEXP_NAME = /\[name\]/i;
+const REGEXP_COMMENTS = /\/\*[\s\S]*?\*\/|([^\\:]|^)\/\/.*$/gm;
+
+const moduleHasNoSource = (module) => {
+  if (typeof module.originalSource !== 'function') return true;
+
+  const source = module.originalSource();
+
+  return Boolean(
+    source &&
+      source
+        .source()
+        // strip all comments
+        .replace(REGEXP_COMMENTS, '')
```

Comment:
It is for `// extracted by mini-css-extract-plugin`?

Reply:
Yep. In order not to remove the modules with valid JS (HMR, locals), to be precise.

Reply:
:+1: I think we don't need regexp here let's check module source on string `// extracted by mini-css-extract-plugin` and all

Reply:
In this case, we should reuse `pluginName` and `// extracted by ${pluginName}` constants between loader and plugin code. Otherwise, we cannot be sure that they will be the same. At the moment, the code is built using only the Babel, and we are not able to require some shared module.

Anyway, I can reluctantly do this, just let me know 😉

Reply:
@irudoy I think we can solve this more simple way - when we replace content on comment we can add property like `module.extracted` and remove only modules what contains this property. Using regexp is bad idea, because other plugin can modify code too and regexp can not works in all cases.

Reply:
Like this?

```javascript
// loader.js
// ...

let resultSource = `// extracted by ${pluginName}`;
let extracted = true;

if (locals) {
    resultSource += `\nmodule.exports = ${JSON.stringify(locals)};`;
    extracted = false;
}

if (options.hmr) {
    resultSource = hotLoader(resultSource, {
        context: this.context,
        options,
        locals,
    });
    extracted = false;
}

this._module.buildMeta = { ...this._module.buildMeta, extracted };

return callback(null, resultSource);
```

Reply:
Yep :+1:

Reply:
Done